zerver/actions/users.py

```diff
@@ -212,7 +214,9 @@ def do_delete_user_preserving_messages(user_profile: UserProfile) -> None:
 
         replacement_user = create_user(
             force_id=user_id,
-            email=f"deleteduser{user_id}@{get_fake_email_domain(realm.host)}",
+            email=Address(
+                username=f"deleteduser{user_id}", domain=get_fake_email_domain(realm.host)
+            ).addr_spec,
```

Comment:
@andersk Just fyi and so you can make sure if this looks good. You added this approach in b945aa3443110cac98736ba33c32ff5899e0b237 and I just failed to use it when adding these functions later, so fixing it up here.